integration/test/test_ffnet_nn_scripts.py

```diff
@@ -259,10 +156,133 @@ def setUpClass(cls):
+            gpu_freq_step = coarse_step
+
+        # GPU Frequency Sweeps for NN Generation - parres
+        self._gpu_experiment_args = SimpleNamespace(
+            node_count=self._node_count,
+            trial_count = 1,
+            cool_off_time=3,
+            parres_cores_per_node=None,
+            parres_gpus_per_node=None,
+            parres_cores_per_rank=1,
+            parres_init_setup=None,
+            parres_exp_setup=None,
+            parres_teardown=None,
+            parres_args=None,
+            output_dir=self._nn_sweep_dir,
+            min_gpu_frequency = gpu_freq_min,
+            max_gpu_frequency = gpu_freq_max,
+            step_gpu_frequency = gpu_freq_step
+        )
+
+        self._app_regions['gpu'] = {'dgemm':'parres_dgemm-0xDEADBEEF',
+                                   'nstream':'parres_nstream-0xDEADBEEF'}
+        app_exec_names = []
+        self._gpu_app_confs = []
+        parres_app_paths = []
+        #Get correct parres executables and set up app conf
+        if util.get_service_config_value('enable_nvml') == '1':
+            app_exec_names = ["dgemm-mpi-cublas", "nstream-mpi-cuda"]
+            self._gpu_app_confs = [parres.create_dgemm_appconf_cuda(mach, self._gpu_experiment_args),
+                         parres.create_nstream_appconf_cuda(mach, self._gpu_experiment_args)]
+        elif util.get_service_config_value('enable_levelzero') == '1':
+            app_exec_names = ["dgemm-onemkl", "nstream-onemkl"]
+            self._gpu_app_confs = [parres.create_dgemm_appconf_oneapi(mach, self._gpu_experiment_args),
+                         parres.create_nstream_appconf_oneapi(mach, self._gpu_experiment_args)]
+        for app in app_exec_names:
+            app_path = os.path.join(parres_basepath, app)
+            if os.path.exists(app_path):
+                parres_app_paths.append(app_path)
+        #If no parres executables exist, error
+        if len(parres_app_paths) == 0:
+            self._do_gpu = False
+            print("Warning: No parres dgemm/nstream executables were found. Cannot test GPU.")
+            return False
+
+    def gen_nn_files(self):
+        # Generate h5s
+        ignore_string = ",".join([reg.split("-")[1] for reg in self._nn_regions_ignore])
+        gen_hdf_from_fsweep.main(self._nn_output_prefix, str(self._nn_sweep_dir), ignore_string)
+        # Generate neural nets
+        gen_neural_net.main(self._nn_trace_hdf, self._nn_out)
+
+        # Generate frequency region map
+        gen_region_parameters.main(self._nn_fmap_out, self._nn_stats_hdf)
+
+        # Clean up unneeded traces/reports
+        #TODO: Replace
+       # for fpath in glob.glob(f"{self._nn_sweep_dir}/*.report"):
+       #     os.remove(fpath)
+       # for fpath in glob.glob(f"{self._nn_sweep_dir}/*.trace-*"):
+       #     os.remove(fpath)
+
+
     ###########
     # Helpers #
     ###########
 
+    def do_gpu(self, mach):
+        parres_basepath = os.path.join(os.path.dirname(
```

Comment:
Might be nice to have some logging to stdout to tell user if GPUs are being used and if not why.